packages/component/src/loadableModulesMap.js

```diff
@@ -0,0 +1,36 @@
+let federatedModuleMap = {};
+const localModuleNames = [];
+
+const isFederated = (loadableCtor) => loadableCtor.resolve().indexOf('webpack/container/remote') === 0;
+
+export const register = (loadableCtor) => {
+  const moduleName = loadableCtor.chunkName();
+  if (isFederated(loadableCtor)) {
+    federatedModuleMap[moduleName] = loadableCtor.importAsync;
+  } else {
+    localModuleNames.push(moduleName);
+  }
+};
+
+/**
+ * Calls importAsync for all the modules listed in moduleNames
+ * Resolves after modules are loaded
+ */
+export const preload = (moduleNames) => {
```

Comment:
it cannot be named as `preload` as well as this file cannot be `loadableModuleMap` as long as it's purely focused on the Federation logic.

Reply:
Initially named it federatedModulesResolver, but then changed the naming. As for preload, it was initially resolve and resolveAll. But then when i started using named imports it got confusing.
Can change it to federatedResolver.js with:
- register
- resolveNamedFederatedModules
- resolveAllFedratedModules